service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -0,0 +1,249 @@
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.DigitalTwins;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.generated.models.DigitalTwinInvokeRootLevelCommandHeaders;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.serialization.DeserializationHelpers;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.serialization.DigitalTwinStringSerializer;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.models.*;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+import com.microsoft.rest.*;
+import com.microsoft.rest.serializer.JacksonAdapter;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.microsoft.azure.sdk.iot.service.digitaltwin.helpers.Tools.*;
+
+public class DigitalTwinAsyncClient {
+    private DigitalTwinsImpl digitalTwin;
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+     DigitalTwinAsyncClient(String connectionString) {
+        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
+        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
+        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+        final SimpleModule stringModule = new SimpleModule("String Serializer");
+        stringModule.addSerializer(new DigitalTwinStringSerializer(String.class, objectMapper));
+
+        JacksonAdapter adapter = new JacksonAdapter();
+        adapter.serializer().registerModule(stringModule);
+        RestClient simpleRestClient = new RestClient.Builder()
+                .withBaseUrl(httpsEndpoint)
+                .withCredentials(new ServiceClientCredentialsProvider(sasTokenProvider))
+                .withResponseBuilderFactory(new ServiceResponseBuilder.Factory())
+                .withSerializerAdapter(adapter)
+                .build();
+
+        IotHubGatewayServiceAPIsImpl protocolLayerClient = new IotHubGatewayServiceAPIsImpl(simpleRestClient);
+        digitalTwin = new DigitalTwinsImpl(simpleRestClient.retrofit(), protocolLayerClient);
+    }
+
+    /**
+     * Creates an implementation instance of {@link DigitalTwins} that is used to invoke the Digital Twin features
+     * @param connectionString The IoTHub connection string
+     * @return DigitalTwinAsyncClient
+     */
+    public static DigitalTwinAsyncClient createFromConnectionString(String connectionString)
+    {
+        return new DigitalTwinAsyncClient(connectionString);
+    }
+
+    /**
+     * Gets a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param clazz The class to deserialize the application/json into.
+     * @param <T> The generic type to deserialize the application/json into.
+     * @return The application/json of the digital twin.
+     */
+    public <T> Observable<T> getDigitalTwin (String digitalTwinId, Class<T> clazz)
```

Comment:
extra space after method name

Reply:
removed here and other places in this class.